module/combat/emotion.py

```diff
@@ -143,5 +149,18 @@ def triggered_bug(self):
 
         return False
 
-    def get_expected_reduce(self, ):
-        pass
+    @cached_property
+    def get_expected_reduce(self):
+        """
+        Returns:
+            int:
+        """
+        if self.config.ENABLE_2X_BOOK and \
+           self.config.COMMAND in ['Main', 'Event', 'War_archives']:
+            return 4
+        else:
+            return 2
+
+    def get_expected_reduce_reset(self):
+        """ Call this method if need to be reset. """
+        del self.__dict__['get_expected_reduce']
```

Comment:
Use `@property` to get expected reduce in real time, not caching it.

Reply:
I can change to this but just wanted to first clear up whether I might be overlooking something. It doesn't look like each call to `self.campaign.run` uses a new forked config. Otherwise it would make sense to make this a `@property` since as you say `ENABLE_2X_BOOK = False` only for a single run then back to `True` after completion. If it still the map cannot support it, changing back to `False`. Repeat until doesn't need to keep resetting that config.